File: ts/src/actions/ModuleConfigurations.ts
import type {ModuleConfigurations} from "@code0-tech/tucana/shared";
import {CodeZeroEvent} from "../events";
import type {Action} from "../action";

export const packetType = "moduleConfigurations";

export function handle(action: Action, data: ModuleConfigurations): void {
    action.configs.update(data.moduleConfigurations);
    action.emit(CodeZeroEvent.moduleUpdated, data);
}

File: py/hercules/actions/module_configurations.py
"""Module configurations handler.

Port of ``src/actions/ModuleConfigurations.ts``.
"""
from __future__ import annotations

from hercules.events import CodeZeroEvent

packet_type = "module_configurations"


def handle(action, data) -> None:
    action.configs.update(data.module_configurations)
    action.emit(CodeZeroEvent.module_updated, data)
